### PR TITLE
Add a couple failure modes to daml new.

### PR DESCRIPTION
**Pull request by associahedron**
Added a couple of failure modes to `daml new`.

Preventing users from confusing project name and template name:

```
$ daml-head new quickstart-java
Template name quickstart-java was given as project name.
Please specify a project name separately, for example:

    daml new myproject quickstart-java

```

Preventing users from creating a project inside another DAML project:

```
$ daml-head new baz
Target directory is inside existing DAML project "/home/work/scratch/bar"
Please specify a new directory outside an existing project.
```

Or inside another DA project:

```
$ daml-head new baz
Target directory is inside existing DA project "/home/work/scratch/bar"
Please convert DA project to DAML using 'daml init':

    daml init /home/work/scratch/bar

Or specify a new directory outside an existing project.
```

Also changed the error message when trying to create a project in an existing folder:

```
$ daml-head new foo
Directory "foo" already exists.
Please specify a new directory, or use 'daml init' instead:

    daml init foo

```

### CODE REVIEW

File: daml-assistant/daml-helper/src/DamlHelper.hs [hunk 1]
```diff
@@ -346,45 +324,127 @@ runInit targetFolderM = do
...
-            , "Please specify a new directory for creating a project."
+            , "Please specify a new directory, or use 'daml init' instead:"
+            , ""
+            , "    daml init " <> escapePath targetFolder
+            , ""
             ]
         exitFailure
+
+    -- Ensure user is not confusing template name with project name.
+    when (isNothing templateNameM && projectName == targetFolder) $ do
```
Comment (cocreature) [inline daml-assistant/daml-helper/src/DamlHelper.hs]: Why do we need the `projectName == targetFolder` check here?

File: daml-assistant/daml-helper/src/DamlHelper.hs [hunk 2]
```diff
@@ -346,45 +324,127 @@ runInit targetFolderM = do
...
+findAscendantWithFile :: FilePath -> FilePath -> IO (Maybe FilePath)
+findAscendantWithFile filename path =
+    findM (\p -> doesFileExist (p </> filename)) (ascendants path)
+
+-- | Escape special characters in a filepath so they can be used as a shell
+-- argument when displaying a suggested command to user. Do not use this to
+-- invoke shell commands directly (there are libraries designed for that).
+escapePath :: FilePath -> FilePath
+escapePath p | isWindows = concat ["\"", p, "\""] -- Windows is a mess
+escapePath p = p >>= \c ->
```
Comment (cocreature) [inline daml-assistant/daml-helper/src/DamlHelper.hs]: I would probably just enclose it in `'` and ignore the case where `'` is in the project name but your choice :smile:

File: daml-assistant/daml-helper/src/DamlHelper.hs (hunk 1 above)
Comment (associahedron) [reply daml-assistant/daml-helper/src/DamlHelper.hs]: My thinking is that if the user passes a targetFolder that isn't a straight up filename (i.e. contains path separators), then they probably know what they're doing.

Comment (associahedron) [reply daml-assistant/daml-helper/src/DamlHelper.hs]: There's a world of difference between `daml new quickstart-java` and `daml new quickstart-java/` :-)

Comment (cocreature) [reply daml-assistant/daml-helper/src/DamlHelper.hs]: Sounds very reasonable, would be nice to capture this thinking in a comment.

File: daml-assistant/daml-helper/src/DamlHelper.hs (hunk 2 above)
Comment (associahedron) [reply daml-assistant/daml-helper/src/DamlHelper.hs]: I find it annoying when tools add quotation marks around paths when they aren't strictly necessary ... so I'll stick with this for now.

File: daml-assistant/daml-helper/src/DamlHelper.hs (hunk 1 above)
Comment (associahedron) [reply daml-assistant/daml-helper/src/DamlHelper.hs]: Ok, i'll add it :)

**Pull request merged**